src/Kokkos_ArithTraits.hpp

```diff
@@ -1348,6 +1377,9 @@ class ArithTraits<__float128> {
   static const bool is_exact = false;
   static const bool is_complex = false;
 
+  static constexpr bool has_infinity = true;
+  static KOKKOS_FORCEINLINE_FUNCTION __float128 infinity() { return 1.0q / 0.0q; }
```

Comment:
Does `__float128` work with CUDA?  I'm pretty sure that it doesn't, but if it does, I would be pleasantly surprised.

Reply:
Nope.  Good catch :)